### PR TITLE
resource pool

### PR DESCRIPTION
**Pull request by noxiouz**
Related to #6 

Blocked by #16 (needs configuration)

### CODE REVIEW

File: insonmnia/miner/server.go [hunk 1]
```diff
@@ -389,6 +394,19 @@ func New(ctx context.Context, config *MinerConfig) (*Miner, error) {
 		cancel()
 		return nil, err
 	}
+
+	var deleter cGroupDeleter
+	if platformSupportCGroups {
```
Comment (antmat) [inline insonmnia/miner/server.go]: This can be simplified if we use NilDeleter

File: insonmnia/miner/cgroup_nonlinux.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+type Resources interface{}
+
+const (
+	platformSupportCGroups = false
+	parentCgroup           = ""
+)
+
+func initializeControlGroup(*Resources) (cGroupDeleter, error) {
+	return nil, fmt.Errorf("%s does not support Control Groups", runtime.GOOS)
```
Comment (antmat) [inline insonmnia/miner/cgroup_nonlinux.go]: How about some kind of NilDeleter instead of error? Looks like it is not very usefull. And we already have a flag indicating cgroups support

File: insonmnia/miner/cgroup_linux.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		}
+	default:
+		return nil, fmt.Errorf("failed to load parent cgroup %s: %v", parentCgroup, err)
+	}
+
+	if res == nil {
+		return control, nil
+	}
+
+	if control.Update((*specs.LinuxResources)(res)); err != nil {
```
Comment (antmat) [inline insonmnia/miner/cgroup_linux.go]: err assignement missed

Comment (noxiouz) [inline insonmnia/miner/cgroup_linux.go]: nice catch

File: insonmnia/miner/cgroup_linux_test.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseResources(t *testing.T) {
+	assert := assert.New(t)
+	defer deleteTestConfigFile()
+	raw := `
+miner:
+  hub_address: 127.0.0.1
```
Comment (3Hren) [inline insonmnia/miner/cgroup_linux_test.go]: This can be omitted.

File: insonmnia/miner/config.go [hunk 5]
```diff
@@ -4,17 +4,21 @@ import (
 	"github.com/jinzhu/configor"
 )
 
-type MinerConfig struct {
```
Comment (3Hren) [inline insonmnia/miner/config.go]: Changes in this file are mostly out of the scope (and is done in https://github.com/sonm-io/core/pull/40), except for adding limitations option.

Also may be rename to `restrictions`?

File: insonmnia/miner/server.go [hunk 6]
```diff
@@ -160,7 +159,7 @@ func (m *Miner) setStatus(status *pb.TaskStatus, id string) {
 func (m *Miner) listenForStatus(statusListener chan pb.TaskStatus_Status, id string) {
 	select {
 	case newStatus := <-statusListener:
-		m.setStatus(&pb.TaskStatus{newStatus}, id)
+		m.setStatus(&pb.TaskStatus{Status: newStatus}, id)
```
Comment (3Hren) [inline insonmnia/miner/server.go]: I'd rather split these changes into a separate PR.

File: insonmnia/miner/config.go (hunk 5 above)
Comment (noxiouz) [inline insonmnia/miner/config.go]: I prefer the same naming as it's in Docker.

Comment (3Hren) [inline insonmnia/miner/config.go]: Got it.
